src/core/Group.cpp

```diff
@@ -187,6 +187,32 @@ QString Group::defaultAutoTypeSequence() const
     return m_data.defaultAutoTypeSequence;
 }
 
+QString Group::effectiveAutoTypeSequence() const
+{
+    QString sequence;
+    bool enableSet = false;
+
+    const Group* group = this;
+    do {
+        if (!enableSet) {
```

Comment:
I'll be honest the way this is coded is very confusing. The while conditions seem like they should be satisfied upfront instead of lumbering through the loop with a return statement embedded in there. Recommend cleaning that up.